### PR TITLE
Integration: Castle deposit and withdraw

### PR DESCRIPTION
**Pull request by adamlawrencium**
Summary of changes:
* Added Castle SDK to support `deposit` and `withdraw`
    * additional supporting instructions `refresh` and `reconcile` were also added
* Added `instructionOption` to support the scenario where a user needs execute additional permissionless instructions
* UI button alongside `ExecuteInstructionButton` that allows user to select an `instructionOption`
* Support for executing adjacent transactions (i.e. attempts to send transactions in the same slot)
* Reorganized instruction list so that nonprotocol-specific ones are at the top

### CODE REVIEW

**Comment by abrzezinski94** [comment]
Hello you need to fix build errors best to run yarn build to check locally 
Zrzut ekranu 2022-04-29 o 08 16 51

File: hooks/useGovernanceAssets.ts [hunk 1]
```diff
@@ -186,6 +186,18 @@ export default function useGovernanceAssets() {
       name: 'Execute Custom Instruction',
       isVisible: canUseAnyInstruction,
     },
+    {
+      id: Instructions.DepositIntoCastle,
+      name: 'Castle: Deposit into Vault',
+      isVisible: true, // TODO - remove hardcoded "true", replace with canUseAnyInstruction
```
Comment (abrzezinski94) [inline hooks/useGovernanceAssets.ts]: Do we want to fix this comment before merge ?

File: utils/validations.tsx [hunk 2]
```diff
@@ -61,8 +61,7 @@ export const tryGetAta = async (
     ASSOCIATED_TOKEN_PROGRAM_ID, // always ASSOCIATED_TOKEN_PROGRAM_ID
     TOKEN_PROGRAM_ID, // always TOKEN_PROGRAM_ID
     mint, // mint
-    owner, // owner
-    true
```
Comment (abrzezinski94) [inline utils/validations.tsx]: Why we want to remove that ?

File: components/instructions/instructionCard.tsx [hunk 3]
```diff
@@ -165,12 +174,19 @@ export default function InstructionCard({
...
-          />
+          <React.Fragment>
+            <ExecuteInstructionButton
+              proposal={proposal}
+              proposalInstruction={proposalInstruction}
+              playing={playing}
+              setPlaying={setPlaying}
+              instructionOption={instructionOption}
+            />
+            <InstructionOptionInput
```
Comment (abrzezinski94) [inline components/instructions/instructionCard.tsx]: if we would like to merge it to main repo we would need to probably hide it based on dao symbol for now

**Comment by abrzezinski94** [comment]
Code looks good just 3 small comments.

File: components/instructions/instructionCard.tsx (hunk 3 above)
Comment (adamlawrencium) [reply components/instructions/instructionCard.tsx]: New soln: Hiding it if castle program appears in the proposal

**Pull request merged**